garcon/decider.py

```diff
@@ -196,7 +197,20 @@ def run(self):
                 running process.
         """
 
-        poll = self.poll()
+        try:
+            poll = self.poll()
+        except Exception as error:
+            # Catch exceptions raised during poll() to avoid a Decider thread
+            # dying & the daemon unable to process subsequent workflows.
+            # on_exception() can be overriden by the flow to send an alert
+            # when such an exception occurs.
+            if self.on_exception:
+                self.on_exception(self, error)
+            else:
+                traceback.print_stack()
+                print('Exception: {}'.format(error))
```

Comment:
went with the todo approach so we can get the fix in a little faster since I'm not sure how much tweaking will need to be done to GarconLogger to get it to work with the Decider